### PR TITLE
TRT-930: Update test to support 2 release query_results.json

### PR DESCRIPTION
**Pull request by deepsm007**
This is the second part to https://github.com/openshift/ci-tools/pull/3562 PR for origin test to support both releases .
Both alerts and disruption tests are updated along with query_results.json.
/cc @dgoodwin @neisw

### CODE REVIEW

File: pkg/invariantlibrary/allowedbackenddisruption/matches_test.go
Comment (dgoodwin) [inline pkg/invariantlibrary/allowedbackenddisruption/matches_test.go:285]: Shouldn't really do comparisons like this on string representations of versions, 4.100 <= 4.14 for example, but in your case, I think != will suffice. If it's not 4.14 specifically, we should be good, as this would not run on any past releases.

Comment (dgoodwin) [reply pkg/invariantlibrary/allowedbackenddisruption/matches_test.go:285]: That said, I think this will immediately start failing when we switch to 5.15 if we haven't updated the data file, which could take up to a week.  That would suck.

Let's just assert <= 2 all the time and be done with it. We'll need to monitor the file ourselves. In future, put a TODO that this can be made to be Equals(2) for 4.15 and onward. Hopefully someone will see it someday.

File: pkg/invariantlibrary/allowedbackenddisruption/matches_test.go
Comment (dgoodwin) [inline pkg/invariantlibrary/allowedbackenddisruption/matches_test.go:324]: Sort isn't going to work if we ever make it to 4.100 :)  But given the above I don't think this function is needed anymore.

File: pkg/invariantlibrary/allowedbackenddisruption/matches_test.go
Comment (neisw) [inline pkg/invariantlibrary/allowedbackenddisruption/matches_test.go:285]: Have already discussed with @deepsm007 to change these to use integer comparison logic and not strings.  Can evaluate using [getMinor](https://github.com/openshift/origin/blob/a4bbe975a9361503a929761b969bf3504225d6ae/pkg/invariantlibrary/historicaldata/next_best_guess.go#L123https://github.com/openshift/origin/blob/a4bbe975a9361503a929761b969bf3504225d6ae/pkg/invariantlibrary/historicaldata/next_best_guess.go#L123) and [getMajor](https://github.com/openshift/origin/blob/a4bbe975a9361503a929761b969bf3504225d6ae/pkg/invariantlibrary/historicaldata/next_best_guess.go#L115C9-L115C9) from next_best_guess.go

**Review by dgoodwin — CHANGES_REQUESTED** [review]

File: pkg/invariantlibrary/allowedbackenddisruption/matches_test.go
Comment (dgoodwin) [reply pkg/invariantlibrary/allowedbackenddisruption/matches_test.go:285]: Just saw Forrest's comment too late. 

There should be a utility out there somewhere for comparing version strings, if you want to keep this check. 

For 4.14, it should be 1, we could test that. For 4.15, it should be 2, but only after the first week or so. The safest thing to do, if you want to keep the test specific, is to enforce 1 for that as well, it will pass when we branch, and then the first time the job submits a PR to origin, we'll see it fail, and someone will have to intervene to update the disruption data.

File: pkg/invariantlibrary/allowedbackenddisruption/matches_test.go
Comment (neisw) [reply pkg/invariantlibrary/allowedbackenddisruption/matches_test.go:324]: Discussed a helper function that could be used here and above that converted the versions and used int comparisons.

File: pkg/invariantlibrary/allowedbackenddisruption/matches_test.go
Comment (neisw) [reply pkg/invariantlibrary/allowedbackenddisruption/matches_test.go:285]: For 4.14 we will actually get 4.13 data, just for non upgrades only since the query contains ` AND (MasterNodesUpdated != "N" OR FromRelease = "")` so the null MasterNodesUpdated records get dropped, except those with no FromRelease.

I'm wondering if we shouldn't manually update the file (run the ci-tools command locally and copy it over) the first time when we are ready and update the test at that time.

File: pkg/invariantlibrary/allowedbackenddisruption/matches_test.go
Comment (neisw) [inline pkg/invariantlibrary/allowedbackenddisruption/matches_test.go:285]: We will actually see the two releases in 4.14, we can look into a manual update of the file in origin when the ci-tools code merges.

**Comment by dgoodwin** [comment]
/approve

Will leave lgtm to Forrest once you guys are happy with how this will land.

**Comment by neisw** [comment]
We still have to work out how it looks at previous release data, current code path will switch from minor upgrade to micro first.  We can discuss needs but seems with this addition if we don't have enough runs we should make an explicit check for the previous release.

File: pkg/invariantlibrary/allowedbackenddisruption/matches_test.go
Comment (neisw) [reply pkg/invariantlibrary/allowedbackenddisruption/matches_test.go:321]: `return historicaldata.GetMajor(releaseSlice[i].Key) > historicaldata.GetMinor(releaseSlice[j].Key)` doesn't look right to me.  We are comparing the Major of i to the Minor of j?  Seems we need to compare both Majors first the Minors.

File: pkg/invariantlibrary/allowedbackenddisruption/matches_test.go
Comment (deepsm007) [reply pkg/invariantlibrary/allowedbackenddisruption/matches_test.go:287]: we won't see 2 releases in 4.14 as 4.13 doesn't have "MASTERNODEUPDATED" column, until we have 4.15 data we should be fine with assert checking for 1 release

File: pkg/invariantlibrary/allowedbackenddisruption/matches_test.go
Comment (deepsm007) [reply pkg/invariantlibrary/allowedbackenddisruption/matches_test.go:323]: Added a helper function instead of GetMajor, GetMinor

File: pkg/invariantlibrary/allowedbackenddisruption/matches_test.go
Comment (neisw) [reply pkg/invariantlibrary/allowedbackenddisruption/matches_test.go:284]: We will see 4.13 data for the non upgrade jobs.  You can verify by running the ci-tools update locally.

File: pkg/invariantlibrary/historicaldata/alert_types.go
Comment (neisw) [inline pkg/invariantlibrary/historicaldata/alert_types.go:91]: Reviewed with Deep, we aren't actually doing anything with the updated matcher here.  Need to consolidate the logic in the loop below for reusability and check it check for results here as well.  Same for disruption.

File: pkg/invariantlibrary/historicaldata/next_best_guess.go
Comment (neisw) [inline pkg/invariantlibrary/historicaldata/next_best_guess.go:224]: Need a fix here and some tests for CurrentReleaseFromMap.

**Comment by deepsm007** [comment]
/payload-aggregate periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-sdn-upgrade 10

**Pull request merged**
